packages/proxy/lib/http/response-middleware.ts

```diff
@@ -345,6 +348,41 @@ const SetInjectionLevel: ResponseMiddleware = function () {
     // We set the header here only for proxied requests that have scripts injected that set the domain.
     // Other proxied requests are ignored.
     this.res.setHeader('Origin-Agent-Cluster', '?0')
+
+    // In order to allow the injected script to run on sites with a CSP header
+    // we must add a generated `nonce` into the response headers
+    const nonce = crypto.randomBytes(16).toString('base64')
+
+    // Iterate through each CSP header
+    cspHeaderNames.forEach((headerName) => {
+      const policyArray = parseCspHeaders(this.res.getHeaders(), headerName)
```

Comment:
`nonce` injection is still happening in `SetInjectionLevel`, but we are going to base it off `this.res.getHeaders()` instead of `this.incomingRes.headers` so that we can pick up the stripped directives in `OmitProblematicHeaders`.